src/arguments.rs

```diff
@@ -0,0 +1,213 @@
+}
+
+#[derive(Parser)]
+#[command(
+    author,
+    version,
+    about,
+    long_about = "Creates a Wasm User-defined aggregate function"
+)]
+pub struct AggregateFunction {
+    #[clap(flatten)]
+    pub shared: SharedFunctionOptions,
+    #[clap(
+        short = 't',
+        long = "type",
+        help = "Return type ~ RETURNS",
+        required = true
+    )]
+    pub return_type: String,
+    #[clap(
+        short,
+        long = "arg",
+        help = "Function argument types. A name will be automatically generated for each argument.",
+        required = true,
+        num_args=1..,
+    )]
+    pub args: Vec<String>,
+    #[clap(
+        short,
+        long = "state",
+        help = "State type ~ WITH STATE",
+        required = true
+    )]
+    pub state_type: String,
+    #[clap(flatten)]
+    pub aggregate_imports: Option<ManualAggregateImports>,
+}
+
+#[derive(Args, Clone)]
+pub struct ManualAggregateImports {
+    #[clap(long, help = "Initialization method ~ INITIALIZE WITH")]
+    pub init: Option<String>,
+    #[clap(long, help = "Update method ~ ITER WITH")]
+    pub iter: Option<String>,
+    #[clap(long, help = "Merge method ~ MERGE WITH")]
+    pub merge: Option<String>,
+    #[clap(long, help = "Terminate method ~ ITER WITH")]
+    pub terminate: Option<String>,
+    #[clap(long, help = "Serialize method ~ SERIALIZE WITH")]
+    pub serialize: Option<String>,
+    #[clap(long, help = "Deserialize method ~ Deserialize WITH")]
+    pub deserialize: Option<String>,
+}
+
+impl ManualAggregateImports {
+    pub fn exit_if_not_valid(&self) {
+        let exit_with = |name| {
+            eprintln!("Aggregate functions require a {name} functions");
+            exit(1);
+        };
+        if self.init.is_none() {
+            exit_with("init");
+        }
+        if self.iter.is_none() {
+            exit_with("iter");
+        }
+        if self.merge.is_none() {
+            exit_with("merge");
+        }
+        if self.terminate.is_none() {
+            exit_with("terminate");
+        }
+
+        if self.serialize.is_none() != self.serialize.is_none() {
+            eprintln!("Aggregate functions require both serialize and deserialize functions");
+            exit(1);
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum ImportSource {
+    Base64(String),
+    Path(PathBuf),
+    Url(Url),
+}
+
+impl ImportSource {
+    /// Unpacks the import source into a string.
+    pub fn unpack(&self) -> String {
+        match self {
+            ImportSource::Base64(base64) => {
+                let decoded = base64::engine::general_purpose::STANDARD
+                    .decode(base64)
+                    .expect("Invalid base64");
+                from_utf8(&decoded).expect("Invalid UTF-8").to_string()
+            }
+            ImportSource::Path(path) => read_to_string(path)
+                .expect("Could not read file")
+                .to_string(),
+            ImportSource::Url(_) => {
```

Comment:
Yes, before I added the support for aggregate member function extraction from a `.wit`, this was a simple `it just works`. This would allow users to provide the github artifacts (or alternative) directly from a URL.

However downloading the `.wit` from URL was a change too many, so decided to defer it.
I can remove the URL import.

Reply:
No need to remove it.  I'd think it'd be a good future enhancement, for sure.